pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -154,17 +144,66 @@ type ControlPlaneInput struct {
 type ContolPlaneJoinInput struct {
 	baseUserData
 
-	CACertHash     string
-	CACert         string
-	CAKey          string
-	BootstrapToken string
-	ELBAddress     string
-	KubeConfig     string
+	CACertHash       string
+	CACert           string
+	CAKey            string
+	EtcdCACert       string
+	EtcdCAKey        string
+	FrontProxyCACert string
+	FrontProxyCAKey  string
+	SaCert           string
+	SaKey            string
+	BootstrapToken   string
+	ELBAddress       string
+}
+
+func (cpi *ControlPlaneInput) isCertMaterialExists() error {
+	if !isKeyPairValid(cpi.CACert, cpi.CAKey) {
+		return errors.New("CA cert material in the ControlPlaneInput is missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.EtcdCACert, cpi.EtcdCAKey) {
+		return errors.New("ETCD CA cert material in the ControlPlaneInput is  missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.FrontProxyCACert, cpi.FrontProxyCAKey) {
+		return errors.New("FrontProxy CA cert material in ControlPlaneInput is  missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.SaCert, cpi.SaKey) {
+		return errors.New("ServiceAccount cert material in ControlPlaneInput is  missing cert/key")
+	}
+
+	return nil
+}
+
+func (cpi *ContolPlaneJoinInput) isCertMaterialExists() error {
+	if !isKeyPairValid(cpi.CACert, cpi.CAKey) {
+		return errors.New("CA cert material in the ContolPlaneJoinInput is  missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.EtcdCACert, cpi.EtcdCAKey) {
+		return errors.New("ETCD cert material in the ContolPlaneJoinInput is  missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.FrontProxyCACert, cpi.FrontProxyCAKey) {
+		return errors.New("FrontProxy cert material in ContolPlaneJoinInput is  missing cert/key")
+	}
+
+	if !isKeyPairValid(cpi.SaCert, cpi.SaKey) {
+		return errors.New("ServiceAccount cert material in ContolPlaneJoinInput is  missing cert/key")
```

Comment:
If we move these to be `KeyPair`s, we could use the method to check and return errors.

Reply:
Agree. However, that will also change the userdata templating that we have.
This PR is already XL and didn't want to cram in more changes. Let's make this better in a follow-up PR. WDYT?